float64 left_pwm
float64 right_pwm
float64 duration